samples/type-ahead-search-adaptive-cards/nodejs/Readme.md

```diff
@@ -0,0 +1,70 @@
+---
+page_type: sample
+description: Demonstrating the feature of type ahead search (static and dynamic) control in Adaptive Cards.
+products:
+- office-teams
+- office
+- office-365
+languages:
+- nodejs
+extensions:
+contentType: samples
+createdDate: "24-12-2021 23:30:17"
+---
+
+# Type ahead search control on Adaptive Cards
+
+This sample shows the feature of type ahead search (static and dynamic) control in Adaptive Cards.
+
+`Static search`
+
+![static search card](Images/staticSearchCard.png)
+
+`Dynamic search`
+
+![dynamic search card](Images/dynamicSearchCard.png)
+
+![dynamic search result](Images/dynamicSearchResult.png)
+
+## Prerequisites
+
+- [NodeJS](https://nodejs.org/en/)
+- [ngrok](https://ngrok.com/) or equivalent tunnelling solution
+
+### 1. Setup for Bot
+In Azure portal, create a [Azure Bot resource](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=csharp%2Caadv2).
+
+- Ensure that you've [enabled the Teams Channel](https://docs.microsoft.com/en-us/azure/bot-service/channel-connect-teams?view=azure-bot-service-4.0)
+
+### 2. Run your bot sample
+1) Clone the repository
+
+    ```bash
+    git clone https://github.com/OfficeDev/Microsoft-Teams-Samples.git
+    ```
+
+2) In the folder where repository is cloned navigate to `samples/type-ahead-search-adaptive-cards/nodejs`
+
+3) Install node modules
+
+   Inside node js folder, open your local terminal and run the below command to install node modules. You can do the same in Visual Studio code terminal by opening the project in Visual Studio code.
+
+    ```bash
+    npm install
+    ```
+4) Run ngrok - point to port 3978
+
+    ```bash
+    ngrok http -host-header=rewrite 3978
+    ```
+5) Open the `.env` configuration file in your project folder (or in Visual Studio Code) and update the `MicrosoftAppId` and `MicrosoftAppPassword` with your app's base url. (Note the ClientId is the AppId created in step 1 (Setup for Bot), the ClientSecret is referred to as the "client secret" in step 1 (Setup for Bot) and you can always create a new client secret anytime.)
+
+6) Run your app
+
+    ```bash
+    npm start
+    ```
+7) Manually update the manifest.json
+    - Edit the `manifest.json` contained in the  `appPackage/` folder to replace with your MicrosoftAppId (that was created in step1.1 and is the same value of MicrosoftAppId in `.env` file) *everywhere* you see the place holder string `{MicrosoftAppId}` (depending on the scenario the Microsoft App Id may occur multiple times in the `manifest.json`)
+    - Zip up the contents of the `appPackage/` folder to create a `manifest.zip`
+    - Upload the `manifest.zip` to Teams (in the left-bottom *Apps* view, click "Upload a custom app")
```

Comment:
Below section is not added in your readme file:
https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/v-pritka/typeahead_search_adaptivecards/samples/type-ahead-search-adaptive-cards/csharp#further-reading

Reply:
added further-reading section